src/jesse_lib.erl

```diff
@@ -106,3 +107,53 @@ is_json_object_empty(Map)
 )
 is_json_object_empty(_) ->
   false.
+
+%% @doc Returns a JSON object in which all lists for
+%% which order is not relevant will be sorted.  In this way, there
+%% will be no differences between objects that contain one of those
+%% lists with the same elements but in different order.  Lists for
+%% which order is relevant, e.g. JSON arrays, keep their original
+%% order and will be considered diffferent if the order is different.
+-spec normalize_and_sort(Value :: any()) -> any().
+normalize_and_sort(Value) ->
+  normalize_and_sort_check_object(Value).
+
+%% This code would look much better if we could use normalize_and_sort_check_object
+%% as a guard expression, but that is not possible.  So we need to check
+%% in every recurssion step, first if the Value is a JSON object with
+%% properties, and in that case call a different function for this values.
+%% @private
+-spec normalize_and_sort_check_object(Value :: any()) -> any().
+normalize_and_sort_check_object(Value) ->
+  case jesse_lib:is_json_object(Value) of
+    true -> normalize_and_sort_object(Value);
+    false -> normalize_and_sort_non_object(Value)
+  end.
+
+%% This function covers the recursion over:
+%% - properties within an object, seen as tuples. In that case, we run
+%%   the normalization/ordering over the values of these properties.
+%% - JSON arrays, seen as lists. In that case, we keep the order of
+%%   the list and run the normalization/ordering over each of the values
+%%   in the list.
+%% - Basic JSON types. In that case, we just return the value.
+%% @private
+-spec normalize_and_sort_non_object(Value :: any()) -> any().
+normalize_and_sort_non_object({Key, Val}) ->
```

Comment:
This `{Key, Val}` clause would be unreachable and can be removed, if you apply my other suggestion for `normalize_and_sort_object`.